exe/Wrapper.hs

```diff
@@ -100,3 +100,25 @@ getRuntimeGhcVersion' cradle = do
         Nothing ->
           die $ "Cradle requires " ++ exe ++ " but couldn't find it" ++ "\n"
            ++ show cradle
+
+-- | Find the cradle that the given File belongs to.
+--
+-- First looks for a "hie.yaml" file in the directory of the file
+-- or one of its parents. If this file is found, the cradle
+-- is read from the config. If this config does not comply to the "hie.yaml"
+-- specification, an error is raised.
+--
+-- If no "hie.yaml" can be found, the implicit config is used.
+-- The implicit config uses different heuristics to determine the type
+-- of the project that may or may not be accurate.
+findLocalCradle :: FilePath -> IO (Cradle Void)
+findLocalCradle fp = do
+  cradleConf <- findCradle fp
+  crdl       <- case cradleConf of
+    Just yaml -> do
+      hPutStrLn stderr $ "Found \"" ++ yaml ++ "\" for \"" ++ fp ++ "\""
+      loadCradle yaml
+    Nothing -> loadImplicitCradle fp
+  hPutStrLn stderr $ "Module \"" ++ fp ++ "\" is loaded by Cradle: " ++ show crdl
+  return crdl
```

Comment:
Dont we need something similar for the `exe/Main.hs`? So, should be put into the library, also for writing tests against it.

Reply:
The corresponding line in Ghcide's exe/Main.hs (src/Development/Ide/Session.hs now) is 
```haskell
cradle <- maybe (loadImplicitCradle $ addTrailingPathSeparator dir) loadCradle hieYaml
```
but it looks like the `hieYaml` is cached in some memo thingy
```haskell
  cradleLoc <- liftIO $ memoIO $ \v -> do
      res <- findCradle v
      -- Sometimes we get C:, sometimes we get c:, and sometimes we get a relative path
      -- try and normalise that
      -- e.g. see https://github.com/digital-asset/ghcide/issues/126
      res' <- traverse makeAbsolute res
      return $ normalise <$> res'
```

I would have liked to have shared that bit but I'm not 100% sure how to abstract it

Reply:
Is that still used? I can imagine that we do not need to cache this.

Reply:
@fendor After looking at it a bit closer I think the memoisation part does make sense, since I presume that shake action can get called/executed (not sure what the terminology is) multiple times on an individual file. And since `findCradle` does a lot of directory traversing we probably don't want to do that every single time